src/chart/draw.js

```diff
@@ -4,13 +4,32 @@ export default function draw(raw_data, processed_data) {
     var context = this;
     let config = this.config;
     let aspect2 = 1 / config.aspect;
+
+    /////////////////////////
+    // Data prep  pipeline //
+    /////////////////////////
+
     //if pre-processing callback, run it now
     this.events.onPreprocess.call(this);
-    //then do normal processing
+
+    // if user passed raw_data to chart.draw(), use that, otherwise use chart.raw_data
     let raw = raw_data ? raw_data : this.raw_data ? this.raw_data : [];
+
+    // warn the user about the perils of "processed_data"
+    if (processed_data) {
+        console.warn(
+            "Drawing the chart using user-defined 'processed_data', this is an experimental, untested feature."
+        );
+    }
+
+    //Call consolidateData - this applies filters from controls and prepares data for each set of marks.
     let data = processed_data || this.consolidateData(raw);
 
-    this.wrap.datum(data);
+    //this.wrap.datum(data);
```

Comment:
What's the rationale behind attaching/not attaching the data to the chart container?

Reply:
There was actually no data being bound here for some reason, so I just took it out.

Reply:
Makes sense to me.  I noticed the data attached to an element upstream of `draw()` is overwritten downstream by this snippet.  I don't necessarily agree with d3 on this point but there's probably a reason.